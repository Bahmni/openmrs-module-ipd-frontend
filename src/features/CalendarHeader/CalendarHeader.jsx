import React from "react";
import "./CalendarHeader.scss";

export default function CalendarHeader() {
  const hours = [...Array(24).keys()];

  return (
    <div className={"calendar-header"}>
      <table>
        <tbody>
          <tr>
            {hours.map((hour) => {
              return (
                <td key={hour}>
                  {hour.toLocaleString("en-US", { minimumIntegerDigits: 2 })}
                  :00
                </td>
              );
            })}
          </tr>
        </tbody>
      </table>
    </div>
  );
}
