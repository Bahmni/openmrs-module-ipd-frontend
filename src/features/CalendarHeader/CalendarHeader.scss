.calendar-header {
  width: max-content;
  position: sticky;
  top: 0;
  border: none;
  background-color: #ededed;
}

.hour-header {
  --additional-borders: 0;
  --border-widths: calc(
    calc(2 + var(--additional-borders)) * var(--line-width)
  );
  --width: calc(calc(2 * var(--half-hour-width)) + var(--border-widths));
  width: var(--width);
  height: 63px;
  color: var(--gray-gray-80, #393939);
  text-align: center;
  font-family: "IBM Plex Sans";
  font-size: 12px;
  letter-spacing: 0.24px;
  padding-bottom: 8px;
  display: grid;
  align-items: end;

  &:last-child {
    --additional-borders: 1;
  }
}
