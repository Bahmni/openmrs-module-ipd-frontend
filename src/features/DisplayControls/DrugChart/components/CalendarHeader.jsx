import React from "react";
import PropTypes from "prop-types";
import "../styles/CalendarHeader.scss";
import data from "../../../../utils/config.json";

export default function CalendarHeader(props) {
  const { currentShiftArray } = props;
  const enable24hour = data.config.drugChart.enable24HourTime;

  return (
    <div className="calendar-header">
      <div style={{ display: "flex" }}>
        {currentShiftArray.map((hour) => {
          const transformedHour = enable24hour ? hour : hour % 12 || 12;
          const formattedHour = `${transformedHour.toLocaleString("en-US", {
            minimumIntegerDigits: 2,
          })}:00`;

          return (
            <div data-testid="hour" key={hour} className={"hour-header"}>
              {formattedHour}
            </div>
          );
        })}
      </div>
    </div>
  );
}
CalendarHeader.propTypes = {
  currentShiftArray: PropTypes.array,
};
