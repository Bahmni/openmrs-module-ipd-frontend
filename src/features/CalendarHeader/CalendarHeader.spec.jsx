import React from "react";
import { render } from "@testing-library/react";
import CalendarHeader from "./CalendarHeader";

describe("CalendarHeader", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<CalendarHeader />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render 24 hours", () => {
    const { getAllByTestId } = render(<CalendarHeader />);
    const hours = getAllByTestId("hour");
    expect(hours.length).toBe(24);
  });
});
