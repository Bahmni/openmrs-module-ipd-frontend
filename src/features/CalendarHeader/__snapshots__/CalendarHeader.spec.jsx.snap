// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="calendar-header"
  >
    <table>
      <tbody>
        <tr>
          <td>
            00:00
          </td>
          <td>
            01:00
          </td>
          <td>
            02:00
          </td>
          <td>
            03:00
          </td>
          <td>
            04:00
          </td>
          <td>
            05:00
          </td>
          <td>
            06:00
          </td>
          <td>
            07:00
          </td>
          <td>
            08:00
          </td>
          <td>
            09:00
          </td>
          <td>
            10:00
          </td>
          <td>
            11:00
          </td>
          <td>
            12:00
          </td>
          <td>
            13:00
          </td>
          <td>
            14:00
          </td>
          <td>
            15:00
          </td>
          <td>
            16:00
          </td>
          <td>
            17:00
          </td>
          <td>
            18:00
          </td>
          <td>
            19:00
          </td>
          <td>
            20:00
          </td>
          <td>
            21:00
          </td>
          <td>
            22:00
          </td>
          <td>
            23:00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
